### PR TITLE
adding thumbnail paginated variant of the carousel and a close event …

### PR DESCRIPTION
**Pull request by natwalker-iap**
…for modals

## Summary of Changes
1. added custom event to modal component to be dispatched when closed
2. added the option to pass thumbnails through to a carousel, to be used instead of pagination dots
3. Updated documentation pages

## Review
Open up the preview site (see the below netlify comment below) and navigate to:
- /components/carousel

## Ticket(s)
FEG-452

## Checklist

The below needs to be done before a pull request can be approved:

- [x] The pull request command has been ran `npm run pull-request`
- [ ] New components work as a Vue component, Web component and as static HTML
- [ ] New components added as an export to src/index.js
- [ ] New components/features have sufficient unit tests
- [ ] New components/features have sufficient documentation
- [ ] New component has dataLayer events added
- [ ] New component is added to the rollup config

## Accesibility check list

- [ ] New components/features are accessible to keyboard users (All links/buttons are tabbable, All content is accessible)
- [ ] New components/features are accessible to non-JS users (All links/buttons are tabbable, All content is accessible), this may have visual differences
- [ ] New components/features have hover, focus and active states on all the links/buttons
- [ ] New components/features work when in dark mode and high contrast mode. Buttons, links and icons should still look like what they are.

### CODE REVIEW

File: assets/ts/components/carousel/carousel.component.ts [hunk 1]
```diff
@@ -61,10 +63,24 @@ class iamCarousel extends HTMLElement {
...
+    }
+
     // populate the pips
     let pips = "";
     for (let i = 1; i <= itemCount; i++) {
-      pips += `<button class="control-${i}" data-slide="${i}" ${i == 1 ? "aria-current":""}>Slide ${i}</button>`;
+      let pipContent = null;
+      let pipClass = '';
+
+      if(thumbnailImages.length){
```
Comment (MartinaAdel) [inline assets/ts/components/carousel/carousel.component.ts:76]:
```suggestion
      if (thumbnailImages.length) {
```

**Review by MartinaAdel — APPROVED** [review]
code looks good, not tested, just tiny suggest

**Review by MartinaAdel — APPROVED** [review]

**Review by whiteias — APPROVED** [review]
Looks good

**Review by arthurrigley — APPROVED** [review]
can't see the preview for some reason, might be updating or outdated 🤔  but code looks fine to me.

**Pull request merged**